Backpack/Snackbar/Classes/BPKSnackbarButton.h

```diff
@@ -0,0 +1,94 @@
+/*
+* Backpack - Skyscanner's Design System
+*
+* Copyright 2018-2020 Skyscanner Ltd
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * Describes the button to show in a snackbar.
+ */
+@interface BPKSnackbarButton: NSObject
+
+/**
+ * The optional visible title that will be displayed.
+ */
+@property(nonatomic, copy, nullable, readonly) NSString *title;
+
+/**
+ * The optional icon that will be displayed
+ */
+@property(nonatomic, strong, nullable, readonly) UIImage *icon;
+
+/**
+ * The accessibility label that VoiceOver will read out for the button.
+ */
+@property(nonatomic, copy, nullable, readonly) NSString *accessibilityLabel;
+
+/**
+ * Whether the receiver will display an icon only button or not.
+ */
+@property(nonatomic, assign, readonly, getter=isIconOnly) BOOL iconOnly;
+
++ (instancetype)new __attribute__((unavailable("Use one of the factory methods")));;
+- (instancetype)init __attribute__((unavailable("Use one of the factory methods")));
+
+/**
+ * A button with only a textual title.
+ *
+ * @param title The title to use.
+ * @return An instance that will display a text only button.
+ */
++ (instancetype)buttonWithTitle:(NSString *)title;
+
+/**
+ * A button with an icon and visible text.
+ *
+ * @param title The title to use.
+ * @param icon The icon to use.
+ * @return An instance that will display a text + icon button.
+ */
++ (instancetype)buttonWithIcon:(UIImage *)icon title:(NSString *)title;
```

Comment:
Is this part of the design, is not supported on Android/RN

Reply:
Dunno but this isn't actually new, the previous API also supported it. By encapsulating this in an object we've only restricted the API to prevent creating icon only buttons that aren't visible to VoiceOver.